src/models/book/dtb/segment.coffee

```diff
@@ -158,6 +158,82 @@ class LYT.Segment
           nextSection.firstSegment().done (next) ->
             next.preloadNext(preloadCount - 1)
 
+  # Read the canvas size from the cartoon page image
+  getCanvasSize: (image) ->
+    result = {}
+    for type in ['height', 'width']
+      # Try reading canvas size from height/width attribute on image
+      if dim = image.attr type
+        result[type] = Number(dim)
+        continue
+      # Try parsing style attribute
+      if dim = image.attr('style').match(type + '\\s*:\\s*(\\d+)\\s*px')
+        result[type] = Number(dim[1])
+        continue
+      # Try reading canvas size from css on the containing div
+      if image.parent().css(type).match(/(\d+)(?:px)?/)
+        result[type] = Number(dim[1])
+        continue
+      else
+        # Finally just give up and use naturalHeight/naturalWidth
+        attr = type.replace /^([a-z])/g, (m, p1) -> 'natural' + p1.toUpperCase()
+        log.warn "render.content: imageDim: no #{type} attribute or css #{type} on image. Falling back to #{attr} which is not known to be cross browser"
+        result[type] = Number(image[attr])
+    result
+
+  getCanvasScale: (canvasSize, imageSize) ->
+    # TODO: We should check that imageSize.width / canvasSize.width == imageSize.height / canvasSize.height
+    #       ...and complain loudly if it isn't the case
+    if canvasSize.width != imageSize.width
+      try
+        imageSize.width / canvasSize.width
```

Comment:
This returns the result right?

Reply:
Yes, the last statement is implicit return in coffeescript. :-1: